app/controllers/individual_market/exchanges/hbx_profiles_controller.rb

```diff
@@ -0,0 +1,162 @@
+        elsif params[:broker].present?
+          agent = Person.find(params[:broker])
+          consumer = Person.find(params[:person])
+          consumer.primary_family.hire_broker_agency(agent.broker_role.id)
+          return ['Broker', agent]
+        elsif params[:assister].present?
+          agent = Person.find(params[:assister])
+          return ['In-Person Assister', agent]
+        end
+        [nil, nil]
+      end
+
+      def process_role_message(role, agent, params)
+        status_text = "Message sent to #{role} #{agent.full_name} <br>"
+        if find_email(agent, role)
+          agent_assistance_messages(params, agent, role)
+        else
+          status_text = "Agent has no email. Please select another."
+        end
+        status_text
+      end
+
+      def find_email(agent, role)
+        if role == 'Broker'
+          agent.try(:broker_role).try(:email).try(:address)
+        else
+          agent.try(:user).try(:email)
+        end
+      end
+
+      def agent_assistance_messages(params, agent, role)
+        # Extract sanitized user details
+        details = extract_and_sanitize_user_details(params)
+
+        # Construct the email body
+        body = construct_message_body(details, params)
+
+        # Retrieve the HbxProfile
+        hbx_profile = HbxProfile.find_by_state_abbreviation(aca_state_abbreviation)
+
+        # Build secure message parameters
+        message_params = {
+          sender_id: hbx_profile.id,
+          parent_message_id: hbx_profile.id,
+          from: 'Plan Shopping Web Portal',
+          to: 'Agent Mailbox',
+          subject: "Please contact #{details[:first_name]} #{details[:last_name]}.",
+          body: body
+        }
+
+        # Send secure messages
+        send_secure_messages(message_params, hbx_profile, agent)
+
+        # Send email notification
+        send_email_notification(
+          agent: agent,
+          role: role,
+          first_name: details[:first_name],
+          full_name: details[:full_name],
+          email: details[:email],
+          person_present: params[:person].present?
+        )
+      end
+
+      def extract_and_sanitize_user_details(params)
+        if params[:person].present?
+          insured = Person.find(params[:person])
+          {
+            first_name: sanitize_html(insured.first_name),
+            last_name: sanitize_html(insured.last_name),
+            full_name: sanitize_html(insured.full_name),
+            email: sanitize_html(insured.emails.last.try(:address) || insured.try(:user).try(:email))
+          }
+        else
+          {
+            first_name: sanitize_html(params[:first_name]),
+            last_name: sanitize_html(params[:last_name]),
+            full_name: sanitize_html("#{params[:first_name]} #{params[:last_name]}"),
+            email: sanitize_html(params[:email])
+          }
+        end
+      end
+
+      def construct_message_body(details, params)
+        if params[:person].present?
```

Comment:
this should be a separate partial probably or something on the front end, I'm approving b/c of business urgency